java/arcs/core/analysis/InformationFlowLabels.kt

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.core.analysis
+
+import java.util.BitSet
+
+/**
+ * An abstract value that is capable of representing a set of set of labels.
+ *
+ * Suppose that we have an universe of labels {A, B, C}. This class may be used to represents sets
+ * of sets of labels like {{A,B}, {A,C}}, {{A}, {B}, {C}}.
+ *
+ * Currently, we use a set of bitsets for the underlying representation, where each bitset
+ * represents a set of labels. Later, we can switch to more efficient representations like
+ * Binary Decision Diagrams (BDD) without changing the interface.
+ */
+data class InformationFlowLabels(
+    private val abstractSetOfLabelSets: AbstractSet<BitSet>
```

Comment:
This name is a bit complicated. How about just "labelSets"? You can document it with an `@param` if you think it's helpful to explain what it's for.

Reply:
Named in `abstractLabelSets` as I wanted to distinguish `AbstractSet<BitSet>`  from `Set<BitSet>`.